source/jormungandr/jormungandr/api.py

```diff
@@ -90,18 +89,22 @@ def add_info_newrelic(response, *args, **kwargs):
     try:
         record_custom_parameter('navitia-request-id', request.id)
 
-        if can_connect_to_database():
-            token = get_token()
+        token = get_token()
+        # No log will be added in newrelic if token is absent in the request
+        if token:
             user = get_user(token=token, abort_if_no_token=False)
-            app_name = get_app_name(token)
             if user:
                 record_custom_parameter('user_id', str(user.id))
-            record_custom_parameter('token_name', app_name)
-
-        record_custom_parameter('version', __version__)
-        coverages = get_used_coverages()
-        if coverages:
-            record_custom_parameter('coverage', coverages[0])
+                # This method verifies database connection and gets object Key only once when cache expires.
+                app_name = get_app_name(token)
+                if app_name:
+                    record_custom_parameter('token_name', app_name)
+
+            record_custom_parameter('version', __version__)
+            # No access to database required
+            coverages = get_used_coverages()
+            if coverages:
+                record_custom_parameter('coverage', coverages[0])
```

Comment:
IMO, we should still track calls without token, as this may cost a little, and we wouldn't see that something is happening without a NR log, no?

Reply:
This is just extra-attribute to the transaction log, so it's gonna be logged anyway, we can forget this comment :+1: